_stbt/core.py

```diff
@@ -719,24 +719,6 @@ def on_warning(self, _bus, message):
     def on_eos_from_source_pipeline(self, _bus, _message):
         if not self.tearing_down:
             warn("Got EOS from source pipeline")
```

Comment:
What do you think about adding:

```python
        self.tell_user_thread(RuntimeError("Received EOS from source pipeline"))
```

Here?

Reply:
Sounds like a good idea. It'd end up raising RuntimeError instead of NoVideo (which inherits from Exception) but I think that's fine. We could change the other NoVideo raised from `get_frame` (on underrun, i.e. no frames received) to also be a RuntimeError.

Reply:
Or we could change them both to NoVideo (i.e. make `get_frame` raise any exceptions from `tell_user_thread` as NoVideo). It turns out that `stbt_core.NoVideo` is part of our `stbt_core` public API.

Reply:
Yeah, that'll make it easier to track.

Reply:
Done:

- EOS will raise `NoVideo: EOS from source pipeline`
- Errors will raise `NoVideo: Error from source pipeline: ...`
- Underrun will raise `NoVideo: No frames received in 10s`
- Errors from the sink pipeline will raise `RuntimeError: Error from sink pipeline: ...`